mongo/integration/retryable_reads_prose_test.go

```diff
@@ -0,0 +1,101 @@
+// Copyright (C) MongoDB, Inc. 2022-present.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License. You may obtain
+// a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+
+package integration
+
+import (
+	"context"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"go.mongodb.org/mongo-driver/bson"
+	"go.mongodb.org/mongo-driver/event"
+	"go.mongodb.org/mongo-driver/mongo/integration/mtest"
+	"go.mongodb.org/mongo-driver/mongo/options"
+)
+
+func TestRetryableReadsProse(t *testing.T) {
+	tpm := newTestPoolMonitor()
+
+	// Client options with MaxPoolSize of 1 and RetryReads used per the test description.
+	// Lower HeartbeatInterval used to speed the test up for any server that uses streaming
+	// heartbeats. Only connect to first host in list for sharded clusters.
+	hosts := mtest.ClusterConnString().Hosts
+	clientOpts := options.Client().SetMaxPoolSize(1).SetRetryReads(true).
+		SetPoolMonitor(tpm.PoolMonitor).SetHeartbeatInterval(500 * time.Millisecond).
+		SetHosts(hosts[:1])
+
+	mtOpts := mtest.NewOptions().ClientOptions(clientOpts).MinServerVersion("4.3")
+	mt := mtest.New(t, mtOpts)
+	defer mt.Close()
+
+	mt.Run("PoolClearedError retryability", func(mt *mtest.T) {
```

Comment:
Optional: Consider adding a comment that references `GODRIVER-1910` and/or the prose test that this implements.

Reply:
I may leave this to GODRIVER-2274 so that we have a consistent format across prose test files.